<launch>
    <!-- Object detection -->
	<arg name="launch_detector" default="true" />
	<arg name="launch_handle_detector" default="true" />

	<include file="$(find rail_object_detector)/launch/detector.launch" if="$(arg launch_detector)">
		<arg name="image_sub_topic_name" value="/kinect/hd/image_color" />
		<arg name="max_desired_publish_freq" value="1.0" />
		<arg name="classnames_filename" value="$(find rail_object_detector)/libs/darknet/data/coco.names" />
		<arg name="cfg_filename" value="$(find rail_object_detector)/libs/darknet/cfg/yolo.cfg" />
		<arg name="weight_filename" value="$(find rail_object_detector)/libs/darknet/yolo.weights" />
	</include>

	<include file="$(find handle_detector)/launch/localization_sensor.launch" if="$(arg launch_handle_detector)">
		<!-- <arg name="as_service" value="true" /> -->
	</include>

	<!-- Object location engine -->
    <node pkg="object_location" type="object_locator" name="object_location_engine" output="screen">    
        <param name="object_detector_service" value="/detector_node/objects_in_scene" />
		<param name="publish_as_tf" value="true" />
		<param name="global_frame" value="map" />
    </node>
</launch>